ckan/views/admin.py

```diff
@@ -143,26 +144,63 @@ class TrashView(MethodView):
     def __init__(self):
         self.deleted_packages = model.Session.query(
             model.Package).filter_by(state=model.State.DELETED)
+        self.deleted_orgs = model.Session.query(model.Group).filter_by(
+            state=model.State.DELETED, is_organization=True)
+        self.deleted_groups = model.Session.query(model.Group).filter_by(
+            state=model.State.DELETED, is_organization=False)
+
+        self.data_type = {
+            u'package': self.deleted_packages,
+            u'organization': self.deleted_orgs,
+            u'group': self.deleted_groups
+        }
 
     def get(self):
-        data = dict(deleted_packages=self.deleted_packages)
-        return base.render(u'admin/trash.html', extra_vars=data)
+        if request.params.get(u'name'):
+            ent_type = request.params.get(u'name').split(u'-')[-1]
+            return base.render(u'admin/snippets/confirm_delete.html',
+                               extra_vars={u'ent_type': ent_type})
+
+        vars = dict(data=self.data_type)
+        return base.render(u'admin/trash.html', extra_vars=vars)
 
     def post(self):
-        if (u'purge-packages' in request.form):
-            for pkg in self.deleted_packages:
-                logic.get_action(u'dataset_purge')(
-                    {u'user': g.user}, {u'id': pkg.id})
+        req_action = request.form.get(u'name', request.params.get(u'name'))
+
+        if u'cancel' in request.form:
+            return h.redirect_to(u'admin.trash')
+
+        if req_action and (u'purge-all' in request.form.get(u'name',
```

Comment:
it's not clear enough - you are checking `req_action`(which represents `name` field from payload) and after that, you are picking `name` field from payload once again? Is it some kind of typo/debug code or i missed something?